File: Strings/2138. Divide a String Into Groups of Size k/2138. Divide a String Into Groups of Size k.js
/**
 * @param {string} s
 * @param {number} k
 * @param {character} fill
 * @return {string[]}
 */
var divideString = function (s, k, fill) {
  const ans = [];
  let i = 0;

  while (i < s.length) {
    let chunk = s.slice(i, i + k);
    if (chunk.length < k) chunk += fill.repeat(k - chunk.length);
    ans.push(chunk);
    i += k;
  }

  return ans;
};

// Time: O(n)
// Space: O(1)

File: Strings/2138. Divide a String Into Groups of Size k/2138. Divide a String Into Groups of Size k.py
class Solution:
    def divideString(self, s: str, k: int, fill: str) -> List[str]:
        ans = []
        i = 0

        while i < len(s):
            chunk = s[i:i + k]
            if len(chunk) < k:
                chunk += fill * (k - len(chunk))
            ans.append(chunk)
            i += k

        return ans

# Time: O(n)
# Space: O(1)
